src/routes/bounce_rules_activity.jsx

```diff
@@ -0,0 +1,228 @@
+import React from "react";
+import { Redirect } from "react-router-dom";
+import ActivityLogContainer from "../components/BounceRulesActivity";
+import { getActivityLog } from "../utils/ruleCalls";
+
+export default class BounceRuleActivityLogPage extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      searchCategory: "Bounce Action",
+      isBounceRulesTab: false,
+      isActivityLogTab: true,
+      searchToken: "",
+      isRedirectingToDetail: false,
+      selectedActivity: {},
+      activityLog: [],
+      currentActivityPageIndex: 1,
+      rulesToShow: 10,
+      pagesToDisplay: 5,
+      filterOptions: [],
+      invalidFilter: false,
+      isFetching: true,
+    };
+    this.logout = this.logout.bind(this);
+    this.updateSearchToken = this.updateSearchToken.bind(this);
+    this.updateSearchCategory = this.updateSearchCategory.bind(this);
+    this.addFilter = this.addFilter.bind(this);
+    this.removeFilter = this.removeFilter.bind(this);
+    this.handleModalClose = this.handleModalClose.bind(this);
+    this.handleActivityTabClicked = this.handleActivityTabClicked.bind(this);
+    this.handleBounceTabClicked = this.handleBounceTabClicked.bind(this);
+    this.updatePageIndex = this.updatePageIndex.bind(this);
+    this.handleActivityClicked = this.handleActivityClicked.bind(this);
+    this.handlePrevClicked = this.handlePrevClicked.bind(this);
+    this.handleNextClicked = this.handleNextClicked.bind(this);
+  }
+
+  async componentDidMount() {
+    const { data: activities } = await getActivityLog();
+    if (activities) {
+      this.setState({
+        isFetching: false,
+        activityLog: activities.reverse(),
+      });
+    }
+  }
+
+  logout() {
+    const { history } = this.props;
+    localStorage.clear();
+    history.push("/");
+  }
+
+  handleModalClose(e) {
+    const { id } = e.currentTarget;
+    this.setState({
+      [id]: false,
+      isInvalidInput: false,
+      selectedRule: {},
+    });
+  }
+
+  updateSearchToken(e) {
+    this.setState({
+      searchToken: e.target.value.toLowerCase(),
+    });
+  }
+
+  updateSearchCategory(e) {
+    this.setState({
+      searchCategory: e.value.toLowerCase(),
+    });
+  }
+
+  filterRules(rules) {
+    const { searchToken } = this.state;
```

Comment:
Will the searchToken always be lowercase as well? we may need to do .toLowerCase() on the searchToken too